src/components/views/elements/MemberEventListSummary.js

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright 2016 OpenMarket Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+const React = require('react');
+const MemberAvatar = require('../avatars/MemberAvatar.js');
+const dispatcher = require("../../../dispatcher");
+
+module.exports = React.createClass({
+    displayName: 'MemberEventListSummary',
+
+    propTypes: {
+        // An array of member events to summarise
+        events: React.PropTypes.array,
+        // The maximum number of names to show in either the join or leave summaries
+        summaryLength: React.PropTypes.number,
+        // The maximum number of avatars to display in the summary
+        avatarsMaxLength: React.PropTypes.number,
+        // The minimum number of events needed to trigger summarisation
+        threshold: React.PropTypes.number,
+        // The function to render events if they are not being summarised
+        renderEvents: React.PropTypes.function,
+        previousEvent: React.PropTypes.object,
+    },
+
+    getInitialState: function() {
+        return {
+            expanded: false,
+        };
+    },
+
+    getDefaultProps: function() {
+        return {
+            summaryLength: 3,
+            threshold: 3,
+            avatarsMaxLength: 5
+        };
+    },
+
+    toggleSummary: function() {
```

Comment:
generally we've been trying to adopt the convention of prefixing private methods with `_`.